nuxeo-services/nuxeo-platform-directory/nuxeo-platform-directory-core/src/main/java/org/nuxeo/ecm/directory/InverseReference.java

```diff
@@ -165,9 +193,27 @@ public void setSourceIdsForTarget(String targetId, List<String> sourceIds) throw
     }
 
     @Override
+    public void setSourceIdsForTarget(String targetId, List<String> sourceIds, Session session)
+            throws DirectoryException {
+        if (readOnly) {
+            return;
+        }
+        checkDualReference();
+        dualReference.setTargetIdsForSource(targetId, sourceIds, session);
+    }
+
+    @Override
     public InverseReference clone() {
-        InverseReference clone = (InverseReference) super.clone();
         // basic fields are already copied by super.clone()
-        return clone;
+        return (InverseReference) super.clone();
+    }
+
+    @Override
+    public void addLinks(String sourceId, List<String> targetIds, Session session) throws DirectoryException {
+        if (readOnly) {
+            return;
+        }
+        checkDualReference();
+        dualReference.addLinks(sourceId, targetIds, session);
```

Comment:
Shouldn't this be `addLinks(targetIds, sourceId, session)`?
So we'd need the symmetrical `addLinks(List<String> sourceIds, String targetId, Session session)` method in the `Reference` interface too, no?